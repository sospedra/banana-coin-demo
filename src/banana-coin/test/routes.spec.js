/* global expect, describe, it */
const routes = require('../routes')
const stubs = require('./stubs')

describe('operations routes', () => {
  it('should return all the operations from getAll', () => {
    const res = stubs.createRes()

    routes.getAll({}, res)

    expect(res.send).toHaveBeenCalledWith([])
    expect(Object.entries(routes.getAll)).toMatchSnapshot()
  })

  it('should create one operation from create', () => {
    const res = stubs.createRes()

    routes.create(stubs.createReq(), res)
    expect(res.send).toHaveBeenCalledWith({ id: 0 })

    routes.create(stubs.createReq(), res)
    expect(res.send).toHaveBeenCalledWith({ id: 1 })

    expect(Object.entries(routes.create)).toMatchSnapshot()
  })
})
