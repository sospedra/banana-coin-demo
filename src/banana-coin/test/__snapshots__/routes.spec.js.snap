// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`operations routes should create one operation from create 1`] = `
Array [
  Array [
    "verb",
    "post",
  ],
  Array [
    "path",
    "/",
  ],
]
`;

exports[`operations routes should return all the operations from getAll 1`] = `
Array [
  Array [
    "verb",
    "get",
  ],
  Array [
    "path",
    "/",
  ],
]
`;
